lib/src/policy/api.ts

```diff
@@ -14,48 +17,49 @@ export async function attributeFQNsAsValues(
   }
   avs.append('withValue.withKeyAccessGrants', 'true');
   avs.append('withValue.withAttribute.withKeyAccessGrants', 'true');
-  const uNoSlash = rstrip(kasUrl, '/');
+  const uNoSlash = rstrip(platformUrl, '/');
   const uNoKas = uNoSlash.endsWith('/kas') ? uNoSlash.slice(0, -4) : uNoSlash;
-  const url = `${uNoKas}/attributes/*/fqn?${avs}`;
-  const req = await authProvider.withCreds({
-    url,
-    headers: {},
-    method: 'GET',
-  });
-  let response: Response;
-  try {
-    response = await fetch(req.url, {
-      mode: 'cors',
-      credentials: 'same-origin',
-      headers: req.headers,
-      redirect: 'follow',
-      referrerPolicy: 'no-referrer',
-    });
-  } catch (e) {
-    throw new NetworkError(`network error [${req.method} ${req.url}]`, e);
-  }
 
-  if (!response.ok) {
-    throw new ServiceError(`${req.method} ${req.url} => ${response.status} ${response.statusText}`);
-  }
+  const platform = new PlatformClient({ authProvider, platformUrl: uNoKas });
 
-  let resp: GetAttributeValuesByFqnsResponse;
+  let response: GetAttributeValuesByFqnsResponse;
   try {
-    resp = (await response.json()) as GetAttributeValuesByFqnsResponse;
+    response = await platform.v1.attributes.getAttributeValuesByFqns({
+      fqns,
+      withValue: {
+        withKeyAccessGrants: true,
+        withAttribute: {
+          withKeyAccessGrants: true,
+        },
+      },
+    });
   } catch (e) {
-    throw new ServiceError(`response parse error [${req.method} ${req.url}]`, e);
+    throw new NetworkError(
+      `[${platformUrl}] [GetAttributeValuesByFqns] ${extractRpcErrorMessage(e)}`
+    );
   }
 
   const values: Value[] = [];
-  for (const [fqn, av] of Object.entries(resp.fqnAttributeValues)) {
-    if (!av.value) {
+  for (const [fqn, av] of Object.entries(response.fqnAttributeValues)) {
+    const value = av.value;
+    if (!value) {
       console.log(`Missing value definition for [${fqn}]; is this a valid attribute?`);
       continue;
     }
-    if (av.attribute && !av.value.attribute) {
-      av.value.attribute = av.attribute;
+    if (value && av.attribute && !value?.attribute) {
+      value.attribute = av.attribute;
     }
-    values.push(av.value);
+    // TODO: update the mapping
```

Comment:
Where is this being used?  I haven't updated the mock server and all the tests are passing. Roundtrip test are passing as well.

The mapping here is not correct: value.attribute, value.grants and value.metadata does not have the right shape and probably needs to be fixed. Where is it used?

Reply:
fixed